### ISSUE TITLE
how to turn on autobatching in python?

### ISSUE THREAD

**Issue by rasoolims** [clab/dynet#588]
I don't see any function for setting it in DynetParams. If I just pass through command line, nothing happens.

**Comment by pmichel31415** [comment]
I haven't added autobatching to the DynetParams object yet but setting the flag in the command line does work.

Are you using `import dynet`?

**Comment by neubig** [comment]
Yes, this should work out-of-the-box by specifying the command line flag.

That being said, this should be added to DynetParams, so I'll mark this as an enhancement.

**Comment by rasoolims** [comment]
When I pass it as a command line argument with the following initialization:
```python
from _gdynet import  *
dyparams = DynetParams()
dyparams.from_args()
dyparams.set_mem(2048)
dyparams.init()
```
I get the following error message:
```
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Dynet does not support both dynamic increasing of memory pool size, and automatic batching or memory checkpointing. If you want to use automatic batching or checkpointing, please pre-allocate enough memory using the --dynet-mem command line option (details http://dynet.readthedocs.io/en/latest/commandline.html).
Aborted (core dumped)
```

**Comment by rasoolims** [comment]
I removed the ```set_mem``` line and passed ```--dynet-mem``` instead and it worked.

**Comment by pmichel31415** [comment]
Hmm interesting so set_mem wasn't working? I'll flag this as a minor bug

**Comment by pmichel31415** [comment]
Well I just checked and it works for me. Let me know if you have more info

**Comment by pmichel31415** [comment]
Fixed by #606
